cmake/enable_lto.cmake

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright 2024-present ScyllaDB
+#
+
+#
+# SPDX-License-Identifier: ScyllaDB-Proprietary
+#
```

Comment:
LicenseRef-ScyllaDB-Source-Available-1.0 ?

Reply:
I wonder if we can add a license check as a GH action.

Reply:
> I wonder if we can add a license check as a GH action.

That'd be great. /cc @yaronkaikov

Reply:
> I wonder if we can add a license check as a GH action.

created at #22065

Reply:
thank you Raphael for pointing this out. should be fixed in the latest revision.